### PR TITLE
feat: Adds support to beagle gRPC library

### PR DESCRIPTION
**Pull request by paulomeurerzup**
Adds support to beagle gRPC library.

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

File: android/processor/src/main/java/br/com/zup/beagle/android/compiler/beaglesetupmanage/TypeElementImplementationManager.kt [hunk 1]
```diff
@@ -56,6 +57,10 @@ internal object TypeElementImplementationManager {
                 val element = propertySpecifications?.httpClient
                 propertySpecifications?.httpClient = manage.manageTypeElement(element, HTTP_CLIENT_HANDLER.className)
             }
+            typeElement.implements(HTTP_CLIENT_FACTORY_HANDLER, processingEnv) -> {
+                val element = propertySpecifications?.httpClientFactory
+                propertySpecifications?.httpClientFactory = manage.manageTypeElement(element, HTTP_CLIENT_FACTORY_HANDLER.className)
```
Comment (botbeaglezup) [inline android/processor/src/main/java/br/com/zup/beagle/android/compiler/beaglesetupmanage/TypeElementImplementationManager.kt:62]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/bf616490f85dadc0bce5cac80f34970f55e1dfb1/android/processor/src/main/java/br/com/zup/beagle/android/compiler/beaglesetupmanage/TypeElementImplementationManager.kt#L62"></span>Line detected that is longer than the defined maximum line length in the code style.</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should add the form local action handler in beagle sdk | br.com.zup.beagle.android.compiler.beaglesdk.FormLocalActionHandlerTest$RegisterFormLocalActionHandler | 0.752|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should add the analytics in beagle sdk | br.com.zup.beagle.android.compiler.beaglesdk.AnalyticsTest$RegisterAnalytics | 0.807|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with duplicate image downloader | br.com.zup.beagle.android.compiler.beaglesdk.BeagleImageDownloaderTest$InvalidImageDownloader | 0.95|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should generate beagle setup | br.com.zup.beagle.android.compiler.BeagleSetupTest$GenerateBeagleSetup | 20.325|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with invalid widget | br.com.zup.beagle.android.compiler.generator.RegisteredWidgetGeneratorTest$InvalidWidgets | 0.966|
Then should show error with invalid widget | br.com.zup.beagle.android.compiler.generator.RegisteredWidgetGeneratorTest$InvalidWidgets | 0.967|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should add the store handler in beagle sdk | br.com.zup.beagle.android.compiler.beaglesdk.StoreHandlerTest$RegisterStoreHandler | 0.676|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with invalid action | br.com.zup.beagle.android.compiler.generator.RegisteredActionGeneratorTest$InvalidActions | 0.744|
Then should show error with invalid action | br.com.zup.beagle.android.compiler.generator.RegisteredActionGeneratorTest$InvalidActions | 0.815|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should create class with list of controllers | br.com.zup.beagle.android.compiler.generator.RegisteredControllerGeneratorTest$Controllers | 0.777|
Then should create class with single controller | br.com.zup.beagle.android.compiler.generator.RegisteredControllerGeneratorTest$Controllers | 0.879|
Then should create class with list of controllers | br.com.zup.beagle.android.compiler.generator.RegisteredControllerGeneratorTest$Controllers | 0.917|
Then should create class with single controller | br.com.zup.beagle.android.compiler.generator.RegisteredControllerGeneratorTest$Controllers | 0.79|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should add the design system in beagle sdk | br.com.zup.beagle.android.compiler.beaglesdk.DesignSystemTest$RegisterDesignSystem | 0.766|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with duplicate logger | br.com.zup.beagle.android.compiler.beaglesdk.BeagleLoggerTest$InvalidLogger | 0.636|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should add the image downloader in beagle sdk | br.com.zup.beagle.android.compiler.beaglesdk.BeagleImageDownloaderTest$Image | 0.744|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with invalid operation | br.com.zup.beagle.android.compiler.generator.RegisteredOperationGeneratorTest$InvalidOperations | 0.781|
Then should show error with invalid operation | br.com.zup.beagle.android.compiler.generator.RegisteredOperationGeneratorTest$InvalidOperations | 0.829|
Then should show error with invalid operation | br.com.zup.beagle.android.compiler.generator.RegisteredOperationGeneratorTest$InvalidOperations | 0.803|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with duplicate http client | br.com.zup.beagle.android.compiler.beaglesdk.HttpClientTest$InvalidHttpClient | 0.726|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should create class with list of widgets | br.com.zup.beagle.android.compiler.generator.RegisteredWidgetGeneratorTest$Widgets | 0.969|
Then should create class with list of widgets | br.com.zup.beagle.android.compiler.generator.RegisteredWidgetGeneratorTest$Widgets | 0.974|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should create class with list of action | br.com.zup.beagle.android.compiler.generator.RegisteredActionGeneratorTest$Actions | 0.751|
Then should create class with list of action | br.com.zup.beagle.android.compiler.generator.RegisteredActionGeneratorTest$Actions | 0.844|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should create class with valid getValidator | br.com.zup.beagle.android.compiler.generator.RegisteredCustomValidatorGeneratorTest$CustomValidators | 1.666|
Then should create class with valid getValidator | br.com.zup.beagle.android.compiler.generator.RegisteredCustomValidatorGeneratorTest$CustomValidators | 1.448|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should create class with map of operations | br.com.zup.beagle.android.compiler.generator.RegisteredOperationGeneratorTest$Operations | 0.986|
Then should create class with map of operations | br.com.zup.beagle.android.compiler.generator.RegisteredOperationGeneratorTest$Operations | 0.807|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should add the http client in beagle sdk | br.com.zup.beagle.android.compiler.beaglesdk.BeagleConfigTest$RegisterBeagleConfig | 0.906|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with duplicate deep link handler | br.com.zup.beagle.android.compiler.beaglesdk.DeepLinkHandlerTest$InvalidDeepLinkHandler | 0.786|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with duplicate analytics | br.com.zup.beagle.android.compiler.beaglesdk.AnalyticsProviderTest$InvalidAnalytics | 0.684|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should add the logger in beagle sdk | br.com.zup.beagle.android.compiler.beaglesdk.BeagleLoggerTest$RegisterLogger | 0.857|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with duplicate store handler | br.com.zup.beagle.android.compiler.beaglesdk.StoreHandlerTest$InvalidStoreHandler | 0.665|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with invalid custom adapter | br.com.zup.beagle.android.compiler.generator.RegisteredCustomAdapterGeneratorTest$InvalidCustomAdapter | 0.923|
Then should show error with invalid custom adapter | br.com.zup.beagle.android.compiler.generator.RegisteredCustomAdapterGeneratorTest$InvalidCustomAdapter | 0.979|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should add the deep link handler in beagle sdk | br.com.zup.beagle.android.compiler.beaglesdk.DeepLinkHandlerTest$RegisterDeepLink | 0.734|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with duplicate url builder | br.com.zup.beagle.android.compiler.beaglesdk.UrlBuilderTest$InvalidUrlBuilder | 0.856|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should create class with valid getAdapter | br.com.zup.beagle.android.compiler.generator.RegisteredCustomAdapterGeneratorTest$CustomAdapters | 1.141|
Then should create class with valid getAdapter | br.com.zup.beagle.android.compiler.generator.RegisteredCustomAdapterGeneratorTest$CustomAdapters | 1.126|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with duplicate design system | br.com.zup.beagle.android.compiler.beaglesdk.DesignSystemTest$InvalidDesignSystem | 0.802|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should add the url builder in beagle sdk | br.com.zup.beagle.android.compiler.beaglesdk.UrlBuilderTest$RegisterUrlBuilder | 0.771|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should add the http client in beagle sdk | br.com.zup.beagle.android.compiler.beaglesdk.HttpClientTest$RegisterHttpClient | 0.723|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should add the analytics in beagle sdk | br.com.zup.beagle.android.compiler.beaglesdk.AnalyticsProviderTest$RegisterAnalytics | 0.686|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with invalid custom validator | br.com.zup.beagle.android.compiler.generator.RegisteredCustomValidatorGeneratorTest$InvalidCustomValidator | 1.218|
Then should show error with invalid custom validator | br.com.zup.beagle.android.compiler.generator.RegisteredCustomValidatorGeneratorTest$InvalidCustomValidator | 1.364|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with duplicate analytics | br.com.zup.beagle.android.compiler.beaglesdk.AnalyticsTest$InvalidAnalytics | 0.667|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Then should show error with duplicate form local action handler | br.com.zup.beagle.android.compiler.beaglesdk.FormLocalActionHandlerTest$InvalidFormLocalActionHandler | 0.762|


<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by matheusribeirozup — APPROVED** [review]

**Comment by matheusribeirozup** [comment]
Maybe the `httpClient` property in `BeagleSdk` should be deprecated.

**Review by matheusribeirozup — APPROVED** [review]

**Review by Tiagoperes — APPROVED** [review]
We need to update both the BeagleDefaults project and the documentation with this change.

**Pull request merged**